fluidBuild.config.cjs

```diff
@@ -90,6 +90,7 @@ module.exports = {
 			dependsOn: ["build:esnext"],
 			script: true,
 		},
+		// new entry for entrypoints + tsc + esnext
```

Comment:
Yes. You should be able to add it now. System doesn't complain when the tasks don't exist.
I am not sure if `build:entrypoints` will need to depend on `^tsc, ^build:esnext, ^api` but I suspect it should depend on some. I would start with:
`"build:entrypoints": ["^build:esnext", "^api-extractor:esnext"],`
then add `"build:entrypoints"` to `tscDependsOn`'s list.

Reply:
What are the actual inputs for this new task? Does it really need the compilation from all dependents? Or does it just need the local build tasks to run? I don't think the ^ is correct.

Reply:
That comment is old - in earlier version of plan we'd update the source entrypoints before local build. Then there would be a single tsc for all. However, that gets pretty complicated; so, we changed to the two build plan. The source generation task should just depend local build now.

Reply:
Why is a build required before this task? Can the source entry points not be found from just the entry point ts files?

Reply:
They can't be found because the new command is generating the entry point .ts files.

Reply:
The new build sequence:
1. build app prior packages and their entry points (all needed .d.ts and .js files should exist)
2. build (tsc) core production package code (index.d.ts and index.js and things it needs now exist)
3. generate/update entry point .ts files (placed in src/entrypoints for clarity; uses index.* to find APIs)
4. build (tsc) entry point .ts files (now production package is complete; lib|dist/entrypoints/*.d.ts and *.js now exist too)

The old build sequence:
1. build app prior packages and their entry points (all needed .d.ts and .js files should exist)
2. build (tsc) core production package code (index.d.ts and index.js and things it needs now exist)
3. generate entry point .d.ts files (now production package is complete; lib|dist/*.d.ts now exist)

Reply:
I thought step 3 could be done with sources alone - not the compilation output.

So each package has two tsconfigs, one for the entrypoints build and one for the regular build, and they need to be mutually exclusive in terms of source files? I.e. TypeScript doesn't know they're connected nor should it ever build the entrypoint ts files (i.e. compile them to js/types) as part of the regular project?

My overall concerns are that this will be a large hit to build perf and it's not clear to me how the build caching should work for the generate:sourceEntrypoints task. What are the inputs and outputs for the build task cache? I thought the only input was src/index.ts, but it sounds like the full set of inputs is src/index.ts, lib+dist/index.js, and lib+dist/index.d.ts, and the outputs are src/entrypoints/*.ts.

Reply:
If I'm correct about the inputs and outputs for the generate entrypoints code, then we can probably use a declarative task and avoid implementing special logic for the fluid-build task.